Formula/php.rb

```diff
@@ -218,7 +222,7 @@ def post_install
     php_ext_dir = opt_prefix/"lib/php"/php_basename
 
     # fix pear config to install outside cellar
-    pear_path = HOMEBREW_PREFIX/"share/pear"
+    pear_path = HOMEBREW_PREFIX/"share/pear@#{php_version}"
```

Comment:
Why do we need to version the main unversioned formula? That typically ends up meaning that we then need to provide automatic migration when e.g. this gets upgraded to 7.3.x

Reply:
Wouldn't it mean the opposite as in if the files are attached to the version itself rather than the latest version it would make more sense ? Otherwise we would have to migrate when the version switches from 7.2 to 7.3 which potentially could break things.

Reply:
The existing files stop working when php is upgraded?

Reply:
You're probably right, I mixed it up with the compiled binaries but they are stored elsewhere. The pear directory only contains the pear php code so it should be reasonably portable between versions.

Reply:
That's fine if we want the user controlled repository to be compatible with the next major/minor release--although we already have the issue that the versioned config `php.ini` doesn't migrate forward. Do you have a suggestion as the the suffix to use? The path without a suffix will always be the repository that upstream installs to make the pear and pecl commands work, and we don't want to conflict with the linked files.